app/components/server_icon/server_icon.test.tsx

```diff
@@ -3,23 +3,25 @@
 
 import React from 'react';
 
-import {renderWithIntlAndTheme} from '@test/intl-test-helper';
+import {renderWithIntlAndTheme, waitFor} from '@test/intl-test-helper';
 
 import Icon from './index';
 
 describe('Server Icon', () => {
-    test('Server Icon Component should match snapshot', () => {
+    test('Server Icon Component should match snapshot', async () => {
         const {toJSON} = renderWithIntlAndTheme(
             <Icon
                 hasUnreads={false}
                 mentionCount={0}
             />,
         );
 
-        expect(toJSON()).toMatchSnapshot();
+        await waitFor(() => {
```

Comment:
For my own knowledge. I see you are doing this in several tests. What is the main reason to do it?

Reply:
this is basically waiting for the badge which is async to render